common/cache/cache_test.go

```diff
@@ -0,0 +1,133 @@
+package cache
+
+import (
+	"testing"
+)
+
+func TestNewLRUCache(t *testing.T) {
+	lruCache := New(5)
+	lruCache.Add("hello", "world")
+	c := lruCache.Contains("hello")
+	if !c {
+		t.Fatal("expected cache to contain \"hello\" key")
+	}
+
+	v := lruCache.Get("hello")
+	if v == nil {
+		t.Fatal("expected cache to contain \"hello\" key")
+	}
+	if v.(string) != "world" {
+		t.Fatal("expected \"hello\" key to contain value \"world\"")
+	}
+
+	r := lruCache.Remove("hello")
+	if !r {
+		t.Fatal("expected \"hello\" key to be removed from cache")
+	}
+
+	v = lruCache.Get("hello")
+	if v != nil {
+		t.Fatal("expected cache to not contain \"hello\" key")
+	}
+}
+
+func TestNewLRUCache_ContainsOrAdd(t *testing.T) {
+	lruCache := New(5)
+
+	f := lruCache.ContainsOrAdd("hello", "world")
+	if f {
+		t.Fatal("expected ContainsOrAdd() to add new key when not found")
+	}
+
+	f = lruCache.ContainsOrAdd("hello", "world")
+	if !f {
```

Comment:
The minorinoist of minor nits
If `f` which equals `true` is `false`, then `fatal` :)
You could remove the funnyness with just `if lruCache.ContainsOrAdd("hello", "world") {` and `if !lruCache.ContainsOrAdd("hello", "world") {`

Reply:
what are you not happy with
f being true & t being false :D